From ecce1532361829a59a87cb8f12afa2af45420453 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Thu, 10 Nov 2022 15:19:44 +0530
Subject: [PATCH] catch IllegalArgumentException from getHarmfulAppWarning

A missing package should not cause system_server crash.

BUG: 253989182
Test: builds
Merged-In: I55fd685280ea6a3d08b9c838b2245ed6dec0aa56
Change-Id: I55fd685280ea6a3d08b9c838b2245ed6dec0aa56
(cherry picked from commit 20e4747134c49501e67a1b0df1fe17f350937787)
Tracked-On: NA
Signed-Off-By: Songchun Fan <schfan@google.com>
---
 .../java/com/android/server/wm/ActivityStartInterceptor.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/wm/ActivityStartInterceptor.java b/services/core/java/com/android/server/wm/ActivityStartInterceptor.java
index a452013bf42a..34220cc40f10 100644
--- a/services/core/java/com/android/server/wm/ActivityStartInterceptor.java
+++ b/services/core/java/com/android/server/wm/ActivityStartInterceptor.java
@@ -380,7 +380,7 @@ class ActivityStartInterceptor {
         try {
             harmfulAppWarning = mService.getPackageManager()
                     .getHarmfulAppWarning(mAInfo.packageName, mUserId);
-        } catch (RemoteException ex) {
+        } catch (RemoteException | IllegalArgumentException ex) {
             return false;
         }
 
-- 
2.17.1

